actors/builtin/storage_market/storage_market_actor.go

```diff
@@ -166,10 +166,10 @@ func (a StorageMarketActor) PublishStorageDeals(rt Runtime, params *PublishStora
 			st.lockBalanceOrAbort(rt, deal.Provider, deal.ProviderBalanceRequirement())
 
 			id := st.generateStorageDealID()
-			onchainDeal := &OnChainDeal{
-				ID:               id,
-				Proposal:         deal,
+			onchainDeal := &DealMeta{
 				SectorStartEpoch: epochUndefined,
+				LastUpdatedEpoch: epochUndefined,
+				SlashEpoch:       epochUndefined,
 			}
 
 			err := deals.Set(id, onchainDeal)
```

Comment:
Blocking: this is setting the meta in the wrong array. At 189 should update `st.Meta` instead.

This would all be less error-prone if the ADT manipulations were inside methods on the state, instead of directly here.